File: apps/web/src/lib/intake.ts
const API_BASE = import.meta.env.VITE_API_URL ?? "http://localhost:8000";

export type InputType = "text" | "chips" | "multi" | "slider" | "range" | "date";

// Full profile payload — mirrors ProfileSchema (PRD §5.4) + ProfileUpsertRequest
export interface ProfilePayload {
  seeker_type: "job" | "housing" | "both";
  name?: string;
  age: number;
  gender: "male" | "female" | "other" | "prefer_not";
  origin_state: string;
  current_city: "Bengaluru";
  native_lang: string;
  languages_spoken: string[];
  migrant_status: "just_moved" | "planning" | "long_term";
  aadhaar_available: boolean;
  // work
  sector?: string;
  skills: string[];
  education: "none" | "primary" | "class10" | "class12" | "diploma" | "grad" | "postgrad";
  years_experience: number;
  employment_status: "unemployed" | "employed" | "underemployed" | "student";
  income_range_inr?: [number, number] | null;
  worker_band: number;
  // housing
  budget_inr?: number | null;
  preferred_areas: string[];
  occupancy_pref?: "single" | "shared" | "dorm" | "any" | null;
  move_in_window_days?: number | null;
}

export interface TurnPrompt {
  en: string;
  hi: string;
  kn: string;
  [lang: string]: string;  // allow dynamic language lookup
}

export interface Turn {
  field: string;
  prompt_i18n: TurnPrompt;
  input_type: InputType;
  options: string[];
  allow_free_text: boolean;
  multi_select: boolean;
}

export interface TurnResponse {
  complete: boolean;
  turn?: Turn;
  current_profile: Record<string, unknown>;
}

export interface FinalizeResponse {
  session_id: string;
  profile: Record<string, unknown>;
}

async function post<T>(path: string, body?: unknown): Promise<T> {
  const res = await fetch(`${API_BASE}${path}`, {
    method: "POST",
    credentials: "include",
    headers: { "Content-Type": "application/json" },
    body: body !== undefined ? JSON.stringify(body) : undefined,
  });
  if (!res.ok) {
    const text = await res.text();
    throw new Error(`${res.status}: ${text}`);
  }
  return res.json() as Promise<T>;
}

export async function startSession(langPref: string = "en") {
  return post<{ session_id: string }>("/api/session/start", { lang_pref: langPref });
}

export async function sendTurn(answer?: { field: string; value: unknown }): Promise<TurnResponse> {
  return post<TurnResponse>("/api/intake/turn", answer ? { answer } : {});
}

export async function finalize(): Promise<FinalizeResponse> {
  return post<FinalizeResponse>("/api/intake/finalize");
}

export async function saveProfile(payload: ProfilePayload): Promise<{ session_id: string; seeker_type: string }> {
  return post("/api/profile", {
    seeker_type: payload.seeker_type,
    profile_json: payload,
  });
}

export interface ResumeParseResponse {
  profile: Partial<ProfilePayload>;
  missing_fields: string[];
  parsed_ok: boolean;
}

export async function parseResume(file: File): Promise<ResumeParseResponse> {
  const form = new FormData();
  form.append("file", file);
  const res = await fetch(`${API_BASE}/api/resume/parse`, {
    method: "POST",
    credentials: "include",
    body: form,
    // Note: do NOT set Content-Type — browser sets it with the boundary automatically
  });
  if (!res.ok) {
    const text = await res.text();
    throw new Error(`${res.status}: ${text}`);
  }
  return res.json() as Promise<ResumeParseResponse>;
}

File: apps/api/schemas/intake.py
"""Pydantic schemas for the intake chat endpoints (§7.3)."""

from __future__ import annotations

import uuid
from typing import Any, Literal, Optional

from pydantic import BaseModel, Field


# ── Request ──────────────────────────────────────────────

class IntakeAnswer(BaseModel):
    """A single user answer submitted during an intake turn."""
    field: str
    value: Any  # str | list[str] | int | bool | tuple[int, int]


class IntakeTurnRequest(BaseModel):
    """Request body for POST /api/intake/turn."""
    session_id: Optional[uuid.UUID] = None
    answer: Optional[IntakeAnswer] = None  # None for the very first turn


# ── Response ─────────────────────────────────────────────

class TurnDefinition(BaseModel):
    """Describes the next question the frontend should render."""
    field: str
    prompt_i18n: dict[str, str]  # {"en": "...", "hi": "...", "kn": "..."}
    input_type: Literal["text", "chips", "multi", "slider", "range", "date"]
    options: list[str] = Field(default_factory=list)
    allow_free_text: bool = False
    multi_select: bool = False


class IntakeTurnResponse(BaseModel):
    """Response body for POST /api/intake/turn."""
    complete: bool = False
    turn: Optional[TurnDefinition] = None
    current_profile: dict[str, Any] = Field(default_factory=dict)


# ── Profile Schema (§5.4) ───────────────────────────────

class ProfileSchema(BaseModel):
    """Full profile schema matching PRD §5.4.

    Used by POST /api/intake/finalize to return the complete
    validated profile for the confirm screen.
    """
    seeker_type: Literal["job", "housing", "both"]
    name: Optional[str] = None
    age: int = 0
    gender: Literal["male", "female", "other", "prefer_not"] = "prefer_not"
    origin_state: str = ""
    current_city: Literal["Bengaluru"] = "Bengaluru"
    native_lang: str = ""
    languages_spoken: list[str] = Field(default_factory=list)
    migrant_status: Literal["just_moved", "planning", "long_term"] = "just_moved"
    aadhaar_available: bool = False

    # Work
    sector: Optional[str] = None
    skills: list[str] = Field(default_factory=list)
    education: Literal[
        "none", "primary", "class10", "class12", "diploma", "grad", "postgrad"
    ] = "none"
    years_experience: int = 0
    employment_status: Literal[
        "unemployed", "employed", "underemployed", "student"
    ] = "unemployed"
    income_range_inr: Optional[list[int]] = None
    worker_band: int = 1  # auto-derived

    # Housing (only when seeker_type != "job")
    budget_inr: Optional[int] = None
    preferred_areas: list[str] = Field(default_factory=list)
    occupancy_pref: Optional[Literal["single", "shared", "dorm", "any"]] = None
    move_in_window_days: Optional[int] = None


class IntakeFinalizeResponse(BaseModel):
    """Response body for POST /api/intake/finalize."""
    profile: ProfileSchema
    session_id: uuid.UUID
